### PR TITLE
Add ⚡️ content to Common user flows - Create a new wallet

### PR DESCRIPTION
**Pull request by danielnordh**
As discussed in issue #470 and the [Google doc](https://docs.google.com/document/d/1Fu-EPdUu9mK3Rsyk_yurWaxUdzGVV1gVD859XMqnQgY/edit?usp=sharing), here is a first PR for the Create a new wallet section.

Preview: https://deploy-preview-497--sad-borg-390916.netlify.app/

### CODE REVIEW

File: guide/designing-products/common-user-flows.md [hunk 1]
```diff
@@ -251,13 +251,19 @@ See the [onboarding section](/guide/onboarding/introduction/) to learn more on t
    layout = "float-left"
 %}
 
-It is essential to distinguish between wallet software and the wallet representing the user's "account." This particular experience may be almost invisible or highly elaborate. While some mobile wallet applications choose to create and secure a wallet during the first launch automatically, other wallets provide extensive configuration wizards.
+By _wallet_, we normally mean the wallet [_application_](/guide/glossary/wallet/#wallet-application). Here we will talk about creating the [_Bitcoin wallet_](/guide/glossary/wallet/#bitcoin-wallet), the private key(s) that the software interacts with to control funds on the Bitcoin blockchain.
```
Comment (Bosch-0) [inline guide/designing-products/common-user-flows.md:254]: This is quite confusing to read, are we creating the wallet software? It may be too far gone for this but I like to think:

Bitcoin Wallet = The application as a whole that constructs / edits / broadcasts transactions.
Bitcoin Account = Private keys generated and used for a specific purpose.

File: guide/designing-products/common-user-flows.md [hunk 2]
```diff
@@ -251,13 +251,19 @@ See the [onboarding section](/guide/onboarding/introduction/) to learn more on t
...
 
 </div>
 
+A good starting point today is an [HD wallet](/guide/glossary/wallet/#hd-wallet) implemented according to [BIP](/guide/glossary/#bip---bitcoin-improvement-proposal)32, 39, 43, 44, 49 and 84. But not all Bitcoin wallets are created this way, as new standards and technologies have emerged over time.
```
Comment (Bosch-0) [inline guide/designing-products/common-user-flows.md:258]: What are the new standards? There is no mention of descriptors in this section!

File: guide/designing-products/common-user-flows.md [hunk 3]
```diff
@@ -251,13 +251,19 @@ See the [onboarding section](/guide/onboarding/introduction/) to learn more on t
...
+
+Some older software may create wallets with outdated technical formats, while others allow users to choose specific formats for their particular needs. Generally, this is difficult to understand for regular users and should either be automatically handled with good default settings, or explained in layman terms. The Wallets Recovery project provides a great overview of different implementations and how nuanced some of the differences are.
```
Comment (Bosch-0) [inline guide/designing-products/common-user-flows.md:260]: Wallets recovery is outdated, we should be pushing for everyone to use descriptors.

File: guide/designing-products/common-user-flows.md [hunk 4]
```diff
@@ -251,13 +251,19 @@ See the [onboarding section](/guide/onboarding/introduction/) to learn more on t
...
 
-One of the most complex users flows in Bitcoin is the creation of multisignature wallets. Multisignature means that there are multiple keys that control the wallet. All keys are required for the creation of the wallet, and users can choose the number of keys required to sign transactions. This is typically done to increase security because it avoids having a single point of failure. For a hacker to get access to 2 out of total of 3 keys that are geographically distributed is much harder than if there is just a single key. While a single user can more easily set up a multisignature wallet with one computer and multiple hardware wallets, it is harder when multiple users hold keys. In that scenario, a file needs to be sent around for each participant to sign. Some wallet software providers hosts servers that seamlessly connect these users. However, this provides another potential attack vector and introduces risk.
+Most modern wallet applications should support the Lightning network in addition to the base layer. While there are different options for how the applications interact with a Lightning network node, an HD wallet works fine for storing the required keys.
```
Comment (Bosch-0) [inline guide/designing-products/common-user-flows.md:264]: Why should they support it? Really depends on the use case. For example I don't see Bitcoin Core even supporting LN. It's not its focus.

File: guide/designing-products/common-user-flows.md [hunk 5]
```diff
@@ -251,13 +251,19 @@ See the [onboarding section](/guide/onboarding/introduction/) to learn more on t
...
+
+Creating multi-key wallets (or multi-signature / multi-sig), can be one of the most complex user flows in Bitcoin. This setup means that several keys control the wallet. All keys are required for the creation of the wallet, and users can choose the number of keys required to sign transactions. This is typically done to increase security. For a hacker to get access to several keys that are geographically distributed is much harder than if there is just a single key. While a single user can somewhat easily set up a multisignature wallet with a computer or mobile device and multiple wallets in their possession, it is harder when different people hold keys. In that scenario, a file needs to be sent around for each participant to sign. Some wallet software providers host servers that seamlessly connect these users. However, this provides another potential attack vector and introduces risk.
```
Comment (Bosch-0) [inline guide/designing-products/common-user-flows.md:266]: Is multi-sig a 'common user flow'? I'd argue not. Lot of content here for something that isn't our primary focus imo.

**Review by Bosch-0 — COMMENTED** [review]
Left some comments, looking good so far!

File: guide/designing-products/common-user-flows.md (hunk 1 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md:254]: Agree it's not entirely satisfying, but it's how we have structured the glossary and how this page referred to it in the past. I suggest we leave it like this for the Lightning changes, and if we want to change to use account (which has other drawbacks imo.) we do that change across all relevant pages.

File: guide/designing-products/common-user-flows.md (hunk 2 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md:258]: The BIP's listed are what I meant by new standards. Are output descriptors captured in any standard that relates to _HD wallets_, or is something that all modern wallet _applications_ should support?

If you were to link to an official spec/description of them would this be it, or something else? https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md

File: guide/designing-products/common-user-flows.md (hunk 3 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md:260]: Feel free to suggest a text change. 
I could imagine cutting the whole second paragraph here to be honest, I don't think it adds a ton.

File: guide/designing-products/common-user-flows.md (hunk 4 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md:264]: I feel we've discussed this. As a community we wan't to push Lightning to be something users don't have to seek out specialist wallets for, hence the default position should be 'support it'. 

There are use cases where it might be less relevant, so I'll amend this slightly, but it should be the exception for a newly started wallet project.

File: guide/designing-products/common-user-flows.md (hunk 5 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md:266]: Agree, I would be up for cutting most or all of this.
As this PR was focused on changes for Lightning I made mostly readability changes.

File: guide/designing-products/common-user-flows.md (hunk 3 above)
Comment (pavlenex) [inline guide/designing-products/common-user-flows.md:260]:
```suggestion
```

To me this seems like a duplicate from the paragraph below?

Screenshot 2021-10-01 at 11 27 02

File: guide/designing-products/common-user-flows.md (hunk 4 above)
Comment (pavlenex) [reply guide/designing-products/common-user-flows.md:264]: I don't think this is a blocker, it says `most` and most should support LN because it's the only way we have now to scale payments.

**Review by pavlenex — COMMENTED** [review]
Just found a minor paragraph I felt read as duplicated.

File: guide/designing-products/common-user-flows.md (hunk 5 above)
Comment (GBKS) [reply guide/designing-products/common-user-flows.md:266]: This was written before a lot of the other content existed. We can toss it (or merge useful parts into other pages), and instead link out to the [Multi-key](https://bitcoin.design/guide/private-key-management/multi-key/), [Creating a new wallet](https://bitcoin.design/guide/onboarding/creating-a-new-wallet/) page as well as the [case studies](https://bitcoin.design/guide/case-studies/) ([Savings account](https://bitcoin.design/guide/case-studies/savings-account/) covers multisig setup well, for example). Having all this other content can really help us turn this page into more of an overview and then just link out.

File: guide/designing-products/common-user-flows.md (hunk 3 above)
Comment (sbddesign) [reply guide/designing-products/common-user-flows.md:260]: Agreed, these paragraphs both tell a similar story

File: guide/designing-products/common-user-flows.md (hunk 1 above)
Comment (sbddesign) [reply guide/designing-products/common-user-flows.md:254]:
```suggestion
By _wallet_, we normally mean the [_wallet application_](/guide/glossary/wallet/#wallet-application). In this context, we will instead talk about creating the [_bitcoin wallet_](/guide/glossary/wallet/#bitcoin-wallet) (the data structure which stores and manages the user's private key(s), which the software interacts with to control funds on the Bitcoin blockchain.
```

Comment (sbddesign) [reply guide/designing-products/common-user-flows.md:254]: I agree with @danielnordh we should keep the terminology consistent. I also agree with @Bosch-0 it's a little confusing to read.

I suggested some sentence structure changes. I think including "in this context" and "instead" will help clarify that we are talking about something different. 
  
Additionally, I borrowed the description "data structure which stores and manages the user's private key(s)" from [chapter 5 of bitcoinbook](https://github.com/bitcoinbook/bitcoinbook/blob/develop/ch05.asciidoc). But that might conflict with the definition in the glossary.
  
Also, I noticed that the main header nav obscures "bitcoin wallet" and "wallet application" when clicking on the glossary links, which may confuse people if they click on those links. It doesn't happen if on the main glossary page, only on the "Wallets" glossary page. I'll open a separate issue for that.

File: guide/designing-products/common-user-flows.md (hunk 2 above)
Comment (sbddesign) [reply guide/designing-products/common-user-flows.md:258]: @Bosch-0 How do output descriptors impact the user experience? What are the significant differences to the user when a wallet does support or does not support this feature? Perhaps this question is outside the scope of this PR discussion, but I'm looking at it through that lens in relation to the Guide

File: guide/designing-products/common-user-flows.md (hunk 4 above)
Comment (sbddesign) [reply guide/designing-products/common-user-flows.md:264]: Agreed, it should be the standard for a new created wallet application in 2021.
  
If we really want to get technical, sure, perhaps there are specific use cases where LN is not needed. But I think for our primary use-case of designing mobile bitcoin wallets, LN should be expected.

File: guide/designing-products/common-user-flows.md (hunk 5 above)
Comment (sbddesign) [reply guide/designing-products/common-user-flows.md:266]: Agreed, I think perhaps just mentioning that multi-sig is a thing and linking to other pages that have more detailed multi-sig information would be sufficient.

**Review by sbddesign — COMMENTED** [review]
Looking good, I left a sentence re-wording suggestion and weighed in on some of the discussions.

File: guide/designing-products/common-user-flows.md (hunk 1 above)
Comment (sbddesign) [reply guide/designing-products/common-user-flows.md:254]: Opened #500 for the issue I describe above with the term being obscured under header

File: guide/designing-products/common-user-flows.md (hunk 2 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md]: I added BIP 380 (output descriptors) to the list, and removed the confusing last sentence.

File: guide/designing-products/common-user-flows.md (hunk 3 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md]: Agreed, my bad. Was not like this in the doc. Fixed now.

File: guide/designing-products/common-user-flows.md (hunk 5 above)
Comment (danielnordh) [reply guide/designing-products/common-user-flows.md]: Replaced with simple mention and link to the multi-key page.

**Review by pavlenex — COMMENTED** [review]
LGTM 🚀

**Review by pavlenex — APPROVED** [review]
LGTM 🚀
